From 8a65fb2101e161bf371eb78bc2962430df97ba96 Mon Sep 17 00:00:00 2001
From: Kaushal M <kaushal@redhat.com>
Date: Wed, 18 Nov 2015 11:21:13 +0530
Subject: [PATCH 137/138] glusterd.service: Ensure rpcbind is started before glusterd

Having the rpcbind.service under the `After` option only ensures that
glusterd.service is started after rpcbind.service if both are
enabled/started at the same time. It doesn't ensure that starting
glusterd.service will start rpcbind.service.

The systemd.unit(5) man page suggests to use both the `Requires` and
`After` options to ensure that rpcbind is started before glusterd,
whenever glusterd is started.

BUG: 1285958
Change-Id: I0f4962fe8da549e789d6bf944e31db0fbdcd6634
Signed-off-by: Kaushal M <kaushal@redhat.com>
Reviewed-upstream-on: http://review.gluster.org/12605
Reviewed-on: https://code.engineering.redhat.com/gerrit/62484
Reviewed-by: Atin Mukherjee <amukherj@redhat.com>
Tested-by: Atin Mukherjee <amukherj@redhat.com>
---
 extras/systemd/glusterd.service.in |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/extras/systemd/glusterd.service.in b/extras/systemd/glusterd.service.in
index 5f962a6..26694cf 100644
--- a/extras/systemd/glusterd.service.in
+++ b/extras/systemd/glusterd.service.in
@@ -1,5 +1,6 @@
 [Unit]
 Description=GlusterFS, a clustered file-system server
+Requires=rpcbind.service
 After=network.target rpcbind.service
 Before=network-online.target
 
-- 
1.7.1

